docs/feature_audio.md

```diff
@@ -1,21 +1,86 @@
 # Audio
 
-Your keyboard can make sounds! If you've got a Planck, Preonic, or basically any AVR keyboard that allows access to certain PWM-capable pins, you can hook up a simple speaker and make it beep. You can use those beeps to indicate layer transitions, modifiers, special keys, or just to play some funky 8bit tunes.
+Your keyboard can make sounds! If you've got a spare pin you can hook up a simple speaker and make it beep. You can use those beeps to indicate layer transitions, modifiers, special keys, or just to play some funky 8bit tunes.
 
-Up to two simultaneous audio voices are supported, one driven by timer 1 and another driven by timer 3.  The following pins can be defined as audio outputs in config.h:
+To activate this feature, add `AUDIO_ENABLE = yes` to your `rules.mk`.
 
-Timer 1:
-`#define B5_AUDIO`
-`#define B6_AUDIO`
-`#define B7_AUDIO`
+## AVR based boards
+Up to two simultaneous tones can be rendered. with one speaker driven by timer 1 and another driven by timer 3.  The following pins can be configured as audio outputs in `config.h`:
 
-Timer 3:
-`#define C4_AUDIO`
-`#define C5_AUDIO`
-`#define C6_AUDIO`
+for the primary speaker, with Timer 3, pick ONE these pins:
```

Comment:
Still not a fan of the config option here, if its changing it should be to something more consistent.

For example:

```c
#define AUDIO_PIN      C4
#define AUDIO_PIN_ALT  B5
```

Reply:
that sadly won't work... since it gets expanded to whatever platform-define C4/B5 will be, ifdef/define switches break... of which there are quite a few to disable unused code-paths - the only solution i currently see is `#define AUDIO_PIN_C4` and the like, which can be used in `#if defined(AUDIO_PIN_C4)`

the nice/consistent solution would have been in the makefile (before the gcc-preprocessor), but that was discouraged :'-(

Reply:
Except it does work, and is already used elsewhere in the code base <https://github.com/qmk/qmk_firmware/blob/master/quantum/backlight/backlight_avr.c#L23>.

Reply:
the chibios cases that give me trouble with these kind of defines would/could be:
- one speaker on A4+Gnd -> need to enable only DACD1
- two separate (and/or different) speakers on A4/A5+Gnd, assigning one as primary, the other as secondary

Reply:
how about this compromise: audio_chibios_dac_[basic|additive] keep the `define AUDIO_PIN_A4` type defines, while audio_[avr|chibios]_pwm_* use `define AUDIO_PIN X0` (where the direct preprocessor comparison works, because B7&co expand to simple hex/int values)

i'd gladly use one consistent type of defines, but currently don't see a good/better solution for the chibios_dac side :-S

Reply:
ha! - found a good(?) solution for the dac side: do more of the AUDIO_PIN checks in c-code instead of the preprocessor - see 677ed4a4a90fac2e74045851616a4ca7413df029

with that the configuration is consitent again over all drivers: AUDIO_PIN A4 - no underscore 

thoughts? does this resolve this thread?

Reply:
marking this sub-thread as resolved - changes are pushed and need another review...